llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8301,6 +8302,9 @@ VPRecipeBuilder::tryToCreatePartialReduction(VPInstruction *Reduction,
          "all accumulators in chain must have same scale factor");
 
   auto *ReductionI = Reduction->getUnderlyingInstr();
+  if (Reduction->getOpcode() == Instruction::FAdd &&
+      !ReductionI->hasAllowReassoc())
+    return nullptr;
```

Comment:
Following on from https://github.com/llvm/llvm-project/pull/159776#discussion_r2556779095, I think this needs to be:
```suggestion
  if (Reduction->getOpcode() == Instruction::FAdd &&
      (!ReductionI->hasAllowReassoc() || !ReductionI->hasAllowContract()))
    return nullptr;
```

Reply:
I believe so. See: https://clang.llvm.org/docs/UsersManual.html#controlling-floating-point-behavior